### ISSUE TITLE
hclwrite printing maps over multiple lines

### ISSUE THREAD

**Issue by rifelpet** [hashicorp/hcl#356]
I'm using hclwrite to print some Terraform resource arguments that are maps. I'm having trouble getting them to span multiple lines using the higher-level hclwrite functions. Similar to #347, I'm wondering if this can be achieved with `SetAttributeValue` or if I'll need to resort to `SetAttributeRaw`.

[This gist](https://gist.github.com/rifelpet/cc42862435c0560157afb8513a71c4e9) shows my three attempts:
1. The first uses `SetAttributeValue` on the entire map value. This is the exact syntax I'm looking for except it is all on one line rather than separate lines per key-value pair.
2. The second uses `AppendNewBlock` and separate `SetAttributeValue`s per pair. This is missing the equals sign as well as quoting around the first key, causing it to be treated as a division expression.
3. The third uses `SetAttributeRaw` on the entire contents of the map. This outputs what I want, I'm just wondering if theres a simpler way.

### HCL Template
The generated HCL from the above gist:

```hcl
resource "aws_elb" "foo1" {
  tags = { "f.o/o" = "bar", foo = "bar" }
}
resource "aws_elb" "foo2" {
  tags {
    f.o / o = "bar"
    foo     = "bar"
  }
}
resource "aws_elb" "foo3" {
  tags = {
    "f.o/o" = "bar"
    "foo"   = "bar"
  }
}
```

The desired HCL:
```hcl
resource "aws_elb" "foo3" {
  tags = {
    "f.o/o" = "foobar"
    foo     = "bar"
  }
}
```
As mentioned, this can be achieved with my third attempt in the gist plus some extra logic around optionally quoting the keys which would be trivial to implement.

Is there an easier way to achieve this with hclwrite or should I continue with `SetAttributeRaw` or perhaps some other utility function like `hclwrite.Format`? Any input would be greatly appreciated. Thanks!

**Comment by leakingtapan** [comment]
I have similar problem but for keep multiline format for list element. I want to append a new element to an existing hcl list programmatically. Any recommendations for how to achieve this?

**Comment by rifelpet** [comment]
@apparentlymart I'm sure you're busy but when you get a chance, do you any thoughts on this?

**Comment by alisdair** [comment]
@rifelpet We recently changed the behaviour of hclwrite in #372 to always generate multiple lines for non-empty maps and objects. This was released in v2.5.0.

I'm closing this issue as I think it's addressed, but feel free to reopen if I'm mistaken.
